kyo-core/shared/src/main/scala/kyo/typeMap.scala

```diff
@@ -10,7 +10,23 @@ extension [A](self: TypeMap[A])
         throw new RuntimeException(s"fatal: kyo.TypeMap of contents [$self] missing value of type: [${t.show}].")
 
     inline def get[B >: A](using inline t: Tag[B]): B =
-        self.getOrElse(t, fatal).asInstanceOf[B]
+        val b = self.getOrElse(t, null)
+        if !isNull(b) then b.asInstanceOf[B]
+        else
+            var sub: B = null.asInstanceOf[B]
+            val it     = self.iterator
+            try
+                while isNull(sub) do
```

Comment:
Using `try/catch` should help with branch prediction, as we should almost always have an element in the iterator due to the type constraints.

Reply:
It might help branch prediction but throwing an exception that doesn't use `NoStackTrace` is much more expensive. I think checking `hasNext` instead would be ideal here.

Reply:
We will throw an exception anyways...Defects/fatal should never occur if someone is not casting.